integration/tests/karpenter/karpenter_test.go

```diff
@@ -101,47 +97,17 @@ var _ = Describe("(Integration) Karpenter", func() {
 })
 
 // not using kubeTest since kubeTest fatals on error, and we don't want that.
-func dumpEventsAndLogs(selectors []string) {
-	config, err := clientcmd.BuildConfigFromFlags("", params.KubeconfigPath)
-	Expect(err).NotTo(HaveOccurred())
-	clientset, err := kubernetes.NewForConfig(config)
-	Expect(err).NotTo(HaveOccurred())
-	// list all events in the Karpenter namespace
-	events := clientset.EventsV1().Events(karpenter.DefaultNamespace)
-	wes, err := events.List(context.Background(), metav1.ListOptions{})
-	Expect(err).NotTo(HaveOccurred())
-	var webhookEvents []eventsv1.Event
-	webhookEvents = append(webhookEvents, wes.Items...)
-	for wes.Continue != "" {
-		wes, err = events.List(context.Background(), metav1.ListOptions{
-			Continue: wes.Continue,
-		})
+func describeKarpenterResources(names []string) {
+	for _, name := range names {
+		cmd := exec.Command("kubectl", "describe", "replicaset", "name", "-n", karpenter.DefaultNamespace)
```

Comment:
The backing ReplicaSet's name will not match the Deployment's name. You'll have to use a label selector to query the RS.

Reply:
Nope, this works fine.

Reply:
```
➜  Run integration tests k describe deployment -n karpenter karpenter-webhook
Name:               karpenter-webhook
Namespace:          karpenter
CreationTimestamp:  Mon, 17 Jan 2022 16:28:38 +0100
Labels:             app.kubernetes.io/managed-by=Helm
Annotations:        deployment.kubernetes.io/revision: 1
                    meta.helm.sh/release-name: karpenter
                    meta.helm.sh/release-namespace: karpenter
Selector:           karpenter=webhook
Replicas:           1 desired | 0 updated | 0 total | 0 available | 1 unavailable
StrategyType:       Recreate
MinReadySeconds:    0
Pod Template:
  Labels:           karpenter=webhook
  Service Account:  karpenter
  Containers:
   webhook:
    Image:      public.ecr.aws/karpenter/webhook:v0.4.3@sha256:576628e37932715cdef2d5bc4bc641a255803b81a2ffb104d674f9f996b10936
    Port:       8443/TCP
    Host Port:  0/TCP
    Limits:
      cpu:     100m
      memory:  50Mi
    Requests:
      cpu:     100m
      memory:  50Mi
    Liveness:  http-get https://:8443/ delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      CLUSTER_NAME:
      CLUSTER_ENDPOINT:
      SYSTEM_NAMESPACE:   (v1:metadata.namespace)
    Mounts:              <none>
  Volumes:               <none>
  Priority Class Name:   system-cluster-critical
Conditions:
  Type             Status  Reason
  ----             ------  ------
  Available        False   MinimumReplicasUnavailable
  ReplicaFailure   True    FailedCreate
  Progressing      False   ProgressDeadlineExceeded
OldReplicaSets:    <none>
NewReplicaSet:     karpenter-webhook-64485c8bd4 (0/1 replicas created)
Events:
  Type    Reason             Age   From                   Message
  ----    ------             ----  ----                   -------
  Normal  ScalingReplicaSet  13m   deployment-controller  Scaled up replica set karpenter-webhook-64485c8bd4 to 1
➜  Run integration tests k describe replicaset -n karpenter karpenter-webhook
Name:           karpenter-webhook-64485c8bd4
Namespace:      karpenter
Selector:       karpenter=webhook,pod-template-hash=64485c8bd4
Labels:         karpenter=webhook
                pod-template-hash=64485c8bd4
Annotations:    deployment.kubernetes.io/desired-replicas: 1
                deployment.kubernetes.io/max-replicas: 1
                deployment.kubernetes.io/revision: 1
                meta.helm.sh/release-name: karpenter
                meta.helm.sh/release-namespace: karpenter
Controlled By:  Deployment/karpenter-webhook
Replicas:       0 current / 1 desired
Pods Status:    0 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           karpenter=webhook
                    pod-template-hash=64485c8bd4
  Service Account:  karpenter
  Containers:
   webhook:
    Image:      public.ecr.aws/karpenter/webhook:v0.4.3@sha256:576628e37932715cdef2d5bc4bc641a255803b81a2ffb104d674f9f996b10936
    Port:       8443/TCP
    Host Port:  0/TCP
    Limits:
      cpu:     100m
      memory:  50Mi
    Requests:
      cpu:     100m
      memory:  50Mi
    Liveness:  http-get https://:8443/ delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      CLUSTER_NAME:
      CLUSTER_ENDPOINT:
      SYSTEM_NAMESPACE:   (v1:metadata.namespace)
    Mounts:              <none>
  Volumes:               <none>
  Priority Class Name:   system-cluster-critical
Conditions:
  Type             Status  Reason
  ----             ------  ------
  ReplicaFailure   True    FailedCreate
Events:
  Type     Reason        Age                   From                   Message
  ----     ------        ----                  ----                   -------
  Warning  FailedCreate  2m26s (x18 over 13m)  replicaset-controller  Error creating: pods "karpenter-webhook-64485c8bd4-" is forbidden: error looking up service account karpenter/karpenter: serviceaccount "karpenter" not found

```